src/devices/machine/upd765.cpp

```diff
@@ -2700,6 +2761,10 @@ void i82072_device::execute_command(int cmd)
 
 		LOGCOMMAND("command motor %s drive %d\n", motor_on ? "on" : "off", fi.id);
 
+		// if we are selecting a different drive, stop the motor on the previously selected drive
+		if (selected_drive != fi.id && flopi[selected_drive].dev && flopi[selected_drive].dev->mon_r() == 0)
```

Comment:
Hey, this `if (` doesn't match the `if(` below it!

Reply:
Also, can you prove it happens?  Case in point, in the atari st (wd1772, external motor control, so different...) if you deselect the drive with the motor running, the motor just keeps running until you re-select the drive and stop it.  Dunno what that chip does, but to you have good hints?

Reply:
The statement doesn't match only because there's no fid local - the effect should be identical, unless I'm missing something obvious?

Regarding the motor control, @cracyc is right - this chip only has one motor output line and the drive select lines are multiplexed, so selecting a different drive will stop the motor on the previously selected one (unless there's external latching of the state, which means the internal motor control can't really do its job). I can't really test this, not having hardware, but the InterPro uses a single floppy drive with id 3, and likes to "stop" it by doing "motor off drive 0", for whatever reason, so this was the conclusion I reached; otherwise drive 3 just keeps spinning indefinitely.

In reality, the i8272/i82072 were a mistake, however maybe another system which used them will turn up one day and we can verify the behaviour.

Reply:
It's also possible that the 87072's motor enable signal (which doesn't exist on the 8272) isn't demultiplexed in the InterPro's single-drive case. Intel's datasheet also suggests using open-collector NAND gates to demultiplex the signal in a two-drive configuration, which would cause deselecting a drive to turn its motor off.